### PR TITLE
Add registerDestination function to the docs

### PR DESCRIPTION
**Pull request by tomfrenken**
## What Has Changed?

This PR adds the `registerDestination` function to the docs.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs/js/features/connectivity/destination.mdx [hunk 1]
```diff
@@ -339,3 +339,30 @@ Custom headers and query parameters, however, overwrite the additional propertie
 **Header names** keep their casing but overwrite other headers independent of casing, e.g. `AUTHORIZATION` overwrites `authorization`.
 This does not apply to query **parameter names**`.
 :::
+
+### Register Destination
+
+With the `registerDestination` function it is possible to create destinations in the environment variables and use them like any other destination.
```
Comment (artemkovalyov) [inline docs/js/features/connectivity/destination.mdx:345]:
```suggestion
With the `registerDestination` function it is possible to create destinations provided via environment variables or by defining them directly in your code. You can then use this destination in the same way as those retrieved from the SAP BTP.
```

Comment (artemkovalyov) [reply docs/js/features/connectivity/destination.mdx:345]: Do I get it correctly you can have a destination defined in your code as a plain JSON or provided as an Env Var?

**Review by artemkovalyov — APPROVED** [review]
It's LGTM. Check my suggestion for clarity please.

File: docs/js/features/connectivity/destination.mdx (hunk 1 above)
Comment (tomfrenken) [reply docs/js/features/connectivity/destination.mdx:345]: No, you always provide it directly as an object, not from the environment variables, the point of this function is to put those destinations into the environment variables.

**Pull request merged**